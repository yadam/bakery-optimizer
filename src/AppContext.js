import PropTypes from 'prop-types';
import React, { useState } from 'react';

export const AppContext = React.createContext();

export const AppContextProvider = ({ children }) => {
  const [breads, setBreads] = useState([]);
  const [customers, setCustomers] = useState([]);
  return (
    <AppContext.Provider value={{ breads, customers, setBreads, setCustomers }}>
      {children}
    </AppContext.Provider>
  );
};

AppContextProvider.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

AppContextProvider.defaultProps = {
  children: undefined,
};

export default { AppContext, AppContextProvider };
