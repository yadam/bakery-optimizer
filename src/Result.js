import React, { memo, useContext, useEffect, useState } from 'react';
import { Box, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { AppContext } from './AppContext';
import { optimize } from './optimizer';

const useStyles = makeStyles((theme) => ({
  root: {
    margin: theme.spacing(6, 0, 3),
  },
  list: {
    padding: theme.spacing(0, 1, 1, 1),
  },
}));

export const Result = memo(() => {
  const { breads, customers } = useContext(AppContext);
  const [result, setResult] = useState({});
  const classes = useStyles();
  useEffect(() => {
    setResult(optimize(breads, customers));
  }, [breads, customers]);

  return (
    <Box className={classes.root}>
      <Typography className={classes.list}>Result:</Typography>
      {result &&
        !Object.keys(result).length &&
        'No solution exists. You will have to call you customers and get one to compromise'}
      {Object.keys(result).map((bread) => (
        <div key={bread}>{`${bread}: ${result[bread]}`}</div>
      ))}
    </Box>
  );
});

export default Result;
