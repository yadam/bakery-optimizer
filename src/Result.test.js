import React from 'react';
import { create } from 'react-test-renderer';
import { AppContext } from './AppContext';
import { Result } from './Result';

const breads = ['sourdough', 'wheat', 'banana'];
const customers = [
  {
    key: 0,
    sourdough: 'round',
  },
  {
    key: 1,
    wheat: 'pan',
    banana: 'round',
  },
  {
    key: 2,
    sourdough: 'pan',
    wheat: 'round',
  },
];

describe('Result', () => {
  it('renders with no results', () => {
    const tree = create(
      <AppContext.Provider value={{ breads: [], customers: [] }}>
        <Result />
      </AppContext.Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with results', () => {
    const tree = create(
      <AppContext.Provider value={{ breads, customers }}>
        <Result />
      </AppContext.Provider>,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
