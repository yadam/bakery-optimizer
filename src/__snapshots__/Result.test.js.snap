// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders with no results 1`] = `
<div
  className="MuiBox-root MuiBox-root-3 makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-list-2 MuiTypography-body1"
  >
    Result:
  </p>
  No solution exists. You will have to call you customers and get one to compromise
</div>
`;

exports[`Result renders with results 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-root-1"
>
  <p
    className="MuiTypography-root makeStyles-list-2 MuiTypography-body1"
  >
    Result:
  </p>
  <div>
    sourdough: round
  </div>
  <div>
    wheat: round
  </div>
  <div>
    banana: round
  </div>
</div>
`;
